source/_integrations/greencell.markdown

```diff
@@ -0,0 +1,226 @@
+---
+title: Greencell
+description: Instructions on how to use the Greencell EVSE integration in Home Assistant.
+ha_category:
+  - Sensor
+ha_iot_class: Local Polling
+ha_release: 2025.10
+ha_codeowners:
+  - '@BrzezowskiGC'
+ha_domain: greencell
+ha_integration_type: integration
+---
+
+The **Greencell** {% term integration %} allows to integrate Greencell electric vehicles charging devices into Home Assistant.
+
+## Supported devices
+
+Currently, the integration supports only [HabuDen](https://greencell.global/habuden) devices.
+
+## Base functionalities
+
+- Monitoring of the device status
+- Measurement of electrical parameters (voltage, current, power)
+- We plan to add charging control process
+
+## Communication
+
+Communication is handled via MQTT, and the integration supports:
+
+- Dynamic state updates and device availability detection
+- Remote start and stop of charging
+- Checking the current charging session and vehicle connection status
+
+{% include integrations/config_flow.md %}
+
+## Integration modes
+
+Greencell offers three levels of integration with Home Assistant to suit different user needs:
+
+| Mode      | Description                                                                                         |
+|:----------|:-----------------------------------------------------------------------------------------------------|
+| **DISABLE** | Integration disabled – the device does not connect to the MQTT broker, and all entities are disabled. |
+| **READ**    | Read only – the device sends measurement data (voltage, current, power), states and ignores commands received on the relevant topic except for the `QUERY` command. Buttons and Number entities are disabled. |
+| **EXECUTE** | Full access – the device sends measurement data and responds to commands (`START`, `STOP`, `SET_CURRENT`, `QUERY`) received on the relevant topic. All supported entities are enabled. |
+
+## Supported entities
+
+### Sensors
+
+- **Charging power** – Instantaneous charging power of the EVSE (W).
+- **Current phase L1** – Current measurement for phase L1 (A).
+- **Current phase L2** – Current measurement for phase L2 (A).
+- **Current phase L3** – Current measurement for phase L3 (A).
+- **Voltage phase L1** – Voltage measurement for phase L1 (V).
+- **Voltage phase L2** – Voltage measurement for phase L2 (V).
+- **Voltage phase L3** – Voltage measurement for phase L3 (V).
+- **EVSE state** – Current state of the EVSE. Possible values:
+  - `UNAVAILABLE`
+  - `IDLE`
+  - `CONNECTED`
+  - `WAITING_FOR_CAR`
+  - `CHARGING`
+  - `FINISHED`
+  - `ERROR_CAR`
+  - `ERROR_EVSE`
+
+## Configure your device using the GC App
+
+This section describes how to configure your device for the Home Assistant integration using the **GC App** from Greencell.
```

Comment:
I'm not sure I understand the installation sequence yet. 
does the Greencell integration need to be added to Home Assistant before I set up Home Assistant in the Greencell app? Or do I need to have the app before I can add the integration to Home Assistant? Is having the app mandatory?